web/ui/static/js/graph.js

```diff
@@ -672,6 +639,136 @@ function escapeHTML(string) {
   });
 }
 
+Prometheus.Page = function() {
+  this.graphs = [];
+};
+
+Prometheus.Page.prototype.init = function() {
+  var graphOptions = this.parseUrl();
+  if (graphOptions.length === 0) {
+    graphOptions.push({});
+  }
+
+  graphOptions.forEach(this.addGraph, this);
+
+  $("#add_graph").click(this.addGraph.bind(this, {}));
+};
+
+Prometheus.Page.prototype.parseUrl = function() {
+  var queryStringBeginning = window.location.href.indexOf('?');
```

Comment:
`window.location.search` would give you the entire query string directly, with leading `?` (empty string if it doesn't exist)... then you also don't need the slicing below.